DEVOPS-693: migrate from conda-lock to pixi

saicheranb opened the pull request:
**DEVOPS-693 - migrate from conda-lock to pixi**

saicheranb committed:
DEVOPS - 693 ::  Update project dependencies and Python version compatibility

- Changed the required Python version from >=3.10,<4.0 to >=3.9,<4.0.
- Expanded the keywords list for better discoverability.
- Added specific dependencies for geoh5py, lasio, numpy, pydantic, and tqdm.
- Removed unnecessary dynamic dependencies and cleaned up the classifiers section.
- Updated the versioning strategy to maintain consistency with conda recipe versioning in tests.
- Refactored the version test to handle edge cases for version comparison.

saicheranb committed:
Update package manager from 'poetry' to 'pixi' in workflow configuration

saicheranb committed:
Remove unused import of Template from version_test.py

sebhmg commented on tests/version_test.py:
```diff
@@ -15,30 +15,36 @@
     project_version = Version(las_geoh5.__version__)
     conda_version = Version(get_conda_recipe_version())
-    assert conda_version.base_version == project_version.base_version
+    if (
+        project_version.base_version == "0.0.0"
+        and conda_version.base_version == "0.0.0"
+    ):
+        return
+
+    if "0.0.0" in (project_version.base_version, conda_version.base_version):
+        assert (
+            project_version.base_version != "0.0.0"
+            or conda_version.base_version != "0.0.0"
+        )
+    else:
+        assert project_version.base_version == conda_version.base_version
```
**issue:** why this change, when the orignal code is working in the other repos?

sebhmg commented on .gitattributes:
```diff
@@ -3,3 +3,5 @@
 *.h5 filter=lfs diff=lfs merge=lfs -text
 *.ipynb filter=lfs diff=lfs merge=lfs -text
 *.min.js filter=lfs diff=lfs merge=lfs -text
+# SCM syntax highlighting & preventing 3-way merges
```
```suggestion

# SCM syntax highlighting & preventing 3-way merges
```

sebhmg commented on .gitignore:
```diff
@@ -142,5 +142,11 @@ dmypy.json
-#version ignore
+# auto-generated version file
 las_geoh5/_version.py
+# not using poetry to lock, but pixi
```
```suggestion

# not using poetry to lock, but pixi
```

sebhmg commented on pixi.toml:
```diff
@@ -0,0 +1,108 @@
+pytest = "*"
+pytest-cov = "*"
+pyyaml = "*"
+jinja2 = "*"
```
jinja is not needed anymore
```suggestion
```

sebhmg commented on pixi.toml:
```diff
@@ -0,0 +1,108 @@
+
+default = {features = ["py310", "mirageo-git", "test", "dev"], solve-group = "default"}
+docs = {features = ["py310", "mirageo-git", "doc"], solve-group = "default"}
+test-docs = {features = ["py310", "mirageo-git", "test", "doc"], solve-group = "default"}
```
not needed here (that was for OMF): there is not test of the docs
```suggestion
```

sebhmg commented on pixi.toml:
```diff
@@ -0,0 +1,108 @@
+
+[feature.test.tasks]
+test = "pytest tests --cov --cov-report=xml"
+_pytest-docs = "pytest docs"
```
**issue:** thre are not tests in docs
```suggestion
```

sebhmg commented on pixi.toml:
```diff
@@ -0,0 +1,108 @@
+build-docs = "sphinx-build -nW -b html docs docs/_build"
+linkcheck-docs = "sphinx-build -nW -b linkcheck docs docs/_build"
+
+[tasks.test-docs]
+depends-on = [{task = "_pytest-docs", environment = "test-docs"}]
+
```
**issue:** irrevelant in this project
```suggestion
```

sebhmg commented on pyproject.toml:
```diff
@@ -13,16 +13,36 @@ build-backend = "poetry_dynamic_versioning.backend"
 license = "MIT"
-keywords = ["geology", "geophysics", "earth sciences", "io", "data", "interoperability"]
+keywords = [
+  "las",
```
👍

sebhmg commented on .pre-commit-config.yaml:
replace call to poetry for pylint by: `pixi run --locked pylint`

sebhmg reviewed (changes requested):
see some suggestions for changes

saicheranb committed:
Refactor pixi.toml and version_test.py

- Removed the test-docs feature from pixi.toml to streamline dependencies.
- Simplified the get_conda_recipe_version function in version_test.py by directly reading and rendering the recipe.yaml file.
- Updated the test_version_is_consistent function to assert version equality directly, removing unnecessary checks for "0.0.0".

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

saicheranb committed:
Fix indentation in test_version_is_consistent and ensure Template import is correctly placed

saicheranb committed:
Update pre-commit configuration and Sphinx documentation settings

- Modify codespell exclusions in .pre-commit-config.yaml
- Upgrade build environment to Ubuntu 24.04 and adjust Python tools in .readthedocs.yml
- Enhance Sphinx configuration in conf.py with additional extensions and Google Analytics support
